app/src/main/java/campus/tech/kakao/map/SearchActivity.kt

```diff
@@ -167,34 +165,78 @@ class SearchActivity : AppCompatActivity() {
                 savedSearchList = db.getAllSavedWords().toMutableList()
                 savedSearchAdapter.savedSearchList = savedSearchList
                 savedSearchAdapter.notifyDataSetChanged()
+
+                saveCoordinates(searchData.x, searchData.y)
+                saveToBottomSheet(searchData.name, searchData.address)
+
+                val intent = Intent(this@SearchActivity, KakaoMapView::class.java)
+                startActivity(intent)
             }
         })
     }
 
-    private fun deleteItem() {
-        savedSearchAdapter.setOnDeleteClickListener(object :
-            SavedSearchAdapter.OnDeleteClickListener {
-            override fun onDeleteClick(position: Int) {
-                val deletedWord = savedSearchAdapter.savedSearchList[position]
-                val wDb = db.writableDatabase
-                wDb.delete(
-                    SearchData.SAVED_SEARCH_TABLE_NAME,
-                    "${SearchData.SAVED_SEARCH_COLUMN_NAME} = ?",
-                    arrayOf(deletedWord)
-                )
+    private fun saveCoordinates(x: Double, y: Double) {
+        val sharedPref = getSharedPreferences("Coordinates", Context.MODE_PRIVATE)
+        with(sharedPref.edit()) {
+            putString("xCoordinate", x.toString())
+            putString("yCoordinate", y.toString())
+            apply()
+        }
+    }
+
+    private fun saveToBottomSheet(name: String, Address: String) {
+        val sharedPref = getSharedPreferences("BottomSheet", Context.MODE_PRIVATE)
+        with(sharedPref.edit()) {
+            putString("name", name)
+            putString("address", Address)
+            apply()
+        }
+    }
 
-                savedSearchAdapter.savedSearchList.removeAt(position)
-                savedSearchAdapter.notifyItemRemoved(position)
+    private fun savedWordClick(){
+        savedSearchAdapter.setOnSavedWordClickListener(object : SavedSearchAdapter.OnSavedWordClickListener {
+            override fun onSavedWordClick(savedWord: String) {
+                filterBySavedWord(savedWord)
+            }
 
+            override fun onDeleteClick(position: Int) {
+                deleteSavedWord(position)
             }
         })
     }
 
+    private fun filterBySavedWord(savedWord: String) {
+        val filteredList = searchDataList.filter { it.name == savedWord }
+        adapter.searchDataList = filteredList
+        adapter.notifyDataSetChanged()
+
+        if (filteredList.isEmpty()) {
+            recyclerView.visibility = View.GONE
+            searchNothing.visibility = View.VISIBLE
+        } else {
+            recyclerView.visibility = View.VISIBLE
+            searchNothing.visibility = View.GONE
+        }
```

Comment:
이건 선택사항이지만 이렇게 정의하는 방법도 있습니다. 
```suggestion
        // import androidx.core.view.isVisible
        recyclerView.isVisible = filteredList.isNotEmpty()
        searchNothing.isVisible = filteredList.isEmpty()
```
하지만 조건이 아주 복잡한 경우에는 if로 조건을 명시해 visibility로 정의하는게 더 나을 때도 있으니 적절히 선택해서 사용해주시면 좋을 것 같아요.